src/audio/component.c

```diff
@@ -240,7 +240,7 @@ int audio_stream_copy(const struct audio_stream *source, uint32_t ioffset,
 	int ssize = audio_stream_sample_bytes(source); /* src fmt == sink fmt */
 	ae_int16x4 *src = (ae_int16x4 *)((int8_t *)audio_stream_get_rptr(source) + ioffset * ssize);
 	ae_int16x4 *dst = (ae_int16x4 *)((int8_t *)audio_stream_get_wptr(sink) + ooffset * ssize);
-	int shorts = samples * ssize >> 1;
+	size_t shorts = samples * ssize >> 1;
```

Comment:
sorry, I don't understand this. Firstly, I think `size_t` is supposed to be used as a number of bytes, whereas here it's a number of `short`s. Secondly, "reducing the risk of a forever loop" - shouldn't such a risk actually be made to be equal to 0? Besides, I don't see how this change affects that risk at all - the loop is over `while (shorts)` so it doesn't seem to be affected by this type change. Thirdly, IIUC `AE_MIN_32_signed(shorts, shorts_copied)` is a signed comparison, so I don't understand why `shorts` now has to be unsigned.

Reply:
so better keep the signed type and use while(short > 0) to avoid the risk, though it shouldn't happen.